### PR TITLE
serial validation set

### PR DESCRIPTION
**Pull request by snurkabill**
tell me your opinion please, it can be done for other neural netoworks (or there could be some abstraction for common methods :) )

### CODE REVIEW

**Comment by mihaelacr** [comment]
There is a problem with this code. The data is no longer shuffled if there is no validation performed. It is important not to change that. 

Please add another method in which the training is deterministic (I guess that is what you are looking for).

**Comment by snurkabill** [comment]
ah, ofc! should be fixed now :)
